R/FilterState.R

```diff
@@ -114,8 +114,8 @@ FilterState <- R6::R6Class( # nolint
       private$dataname <- dataname
       private$varname <- varname
       private$selected <- reactiveVal()
-      private$keep_na <- reactiveVal(keep_na)
-      private$keep_inf <- reactiveVal(keep_inf)
+      private$keep_na <- if (is.null(keep_na)) reactiveVal(TRUE) else reactiveVal(keep_na)
+      private$keep_inf <- if (is.null(keep_inf)) reactiveVal(TRUE) else reactiveVal(keep_inf)
```

Comment:
We shouldn't return `keep_inf = TRUE` in case when there is no `Inf` in `x`. Also we shouldn't return `is.infinite()` for variables other than numerics.

So, `keep_na` and `keep_inf` should be `TRUE` only if they occur in `x`